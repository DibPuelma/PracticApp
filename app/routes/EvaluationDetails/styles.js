import {
  StyleSheet,
  Dimensions
} from 'react-native';

module.exports = StyleSheet.create({
  background: {
    flex: 1,
    marginLeft: 10,
    marginTop: 72,
    marginRight: 10,
    marginBottom: 10,
    elevation: 2,
    justifyContent: 'center',
    alignItems: 'center',
    borderColor: '#C6C6C6',
    borderWidth: 1
  },
  container: {
    alignItems: 'center'
  },
  iconAndTextContainer: {
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  logo: {
    width: 80,
    height: 80,
    margin: 5
  },
  storeName: {
    fontSize: 25,
    marginBottom: 10,
    fontWeight: 'bold',
    textAlign: 'right',
    borderBottomColor: '#c6c6c6',
    borderBottomWidth: 1,
    borderStyle: 'solid'
  },
  questionContainer: {
    alignItems: 'center',
    marginTop: 5,
    marginBottom: 5,
    paddingBottom: 20,
    borderBottomColor: '#e6e6e6',
    borderBottomWidth: 1,
    borderStyle: 'solid'
  },
  question: {
    fontSize: 20,
    textAlign: 'center',
    paddingLeft: 10,
    paddingRight: 10,
    flex:4
  },
  answer: {
    marginTop: 10
  },
  textAnswer: {
    textAlign: 'center',
    fontSize: 18
  },
  bigTextAnswer: {
    textAlign: 'center',
    fontSize: 24
  }
})
