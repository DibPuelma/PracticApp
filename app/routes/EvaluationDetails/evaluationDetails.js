import React, { Component } from 'react';
import {
  Image,
  Text,
  View,
  ScrollView,
  ActivityIndicator
} from 'react-native';

import MyStarRating from '../../components/MyStarRating/myStarRating';
import backButtonHandler from '../../lib/backButtonHandler';
import styles from './styles';
import LoadingSpinner from '../../components/LoadingSpinner/loadingSpinner';
import settings from '../../config/settings';

export default class EvaluationDetails extends Component {
  constructor(props){
    super(props);
    this.singletonBackButtonHandler = backButtonHandler.getInstance();
    this._backToPrevious = this._backToPrevious.bind(this);
    var uri = settings.EVALUATIONS_DETAILS_REQUEST.replace(":user_id", props.user.id).replace(":answered_poll_id", props.sellPointData.id);
    this.state = {
      ready: false,
      uri: uri
    };
  }

  componentDidMount(){
    console.log("uri: " + this.state.uri);
    fetch(this.state.uri, {
      method: 'GET',
      headers: {
        'Accept': 'application/json',
        'Content-Type': 'application/json',
      }
    })
    .then((response) => response.json())
    .then((responseJson) => {
      this.setState({answeredPoll: responseJson});
      this.setState({ready: true});
    })
    .catch((error) => {
      console.error(error);
    });
  }

  componentWillMount(){
    this.singletonBackButtonHandler.addBackEvent(this._backToPrevious);
  }

  componentWillUnmount() {
    this.singletonBackButtonHandler.removeBackEvent(this._backToPrevious);
  }

  _backToPrevious() {
    this.props.navigator.pop();
    return true; // This is important to prevent multiple calls
  }

  render(){
    if(!this.state.ready) {
      return (
        <LoadingSpinner/>
      );
    }
    else {
      return(
        <View style={styles.background}>
        <ScrollView contentContainerStyle={styles.container}>
        <Image style={styles.logo} source={{uri:this.props.sellPointData.logo}} />
        <Text style={styles.storeName}> {this.props.sellPointData.name} </Text>
        {this.state.answeredPoll.Answers.map((answer, i) => {
            console.log("####################  ANSWER");
            console.log(answer);
            console.log("####################  ANSWER");
          if(answer.number_value !== null){
            return (
              <View key={i} style={styles.questionContainer}>
              <Text style={styles.question}> {answer.Question.text} </Text>
              <MyStarRating isDisabled={true} style={styles.starAnswer} initialRate={answer.number_value}/>
              </View>
            )
          }
          else if(answer.string_value !== null){
            return (
              <View key={i} style={styles.questionContainer}>
              <Text style={styles.question}> {answer.Question.text} </Text>
              <Text style={[styles.answer, styles.textAnswer]}> "{answer.string_value}"</Text>
              </View>
            )
          }
          else if(answer.boolean_value !== null){
            return (
              <View key={i} style={styles.questionContainer}>
              <Text style={styles.question}> {answer.Question.text} </Text>
              <Text style={[styles.answer, styles.bigTextAnswer]}> "{answer.boolean_value ? "Sí" : "No"}"</Text>
              </View>
            )
          }
          else{
            return(
              <View key={i} style={styles.questionContainer}>
              <Text style={styles.question}> {answer.Question.text} </Text>
              <Text style={[styles.answer, styles.bigTextAnswer]}> "{answer.PossibleOption.value}" </Text>
              </View>
            )
          }
        }
      )}

      </ScrollView>
      </View>
    );
  }
}
}
