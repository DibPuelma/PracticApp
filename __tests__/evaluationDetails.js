import 'react-native';
import React from 'react';
import EvaluationDetails from '../app/routes/EvaluationDetails/evaluationDetails'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

evaluationData = {
  "questions": [
    {"question":"¿La tienda está ordenada?", "stars": 2},
    {"question":"¿Le gustan los productos de la tienda?", "stars":3},
    {"question":"¿Qué nota le pondría a la tienda en general?", "stars":2}
  ]

}
it('renders correctly', () => {
  const tree = renderer.create(
    <EvaluationDetails evaluationData={evaluationData}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
