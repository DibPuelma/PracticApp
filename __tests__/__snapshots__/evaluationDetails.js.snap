exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E6E6E6",
      "borderColor": "#C6C6C6",
      "borderWidth": 1,
      "elevation": 3,
      "flex": 1,
      "justifyContent": "center",
      "marginBottom": 10,
      "marginLeft": 10,
      "marginRight": 10,
      "marginTop": 72,
    }
  }>
  <ScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E6E6E6",
      }
    }>
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "height": 80,
          "margin": 10,
          "width": 80,
        }
      } />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
          "marginTop": 5,
          "textAlign": "right",
        }
      }>
       
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "paddingLeft": 10,
            "paddingRight": 10,
            "textAlign": "center",
          }
        }>
         
        ¿La tienda está ordenada?
         
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gold",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gold",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "paddingLeft": 10,
            "paddingRight": 10,
            "textAlign": "center",
          }
        }>
         
        ¿Le gustan los productos de la tienda?
         
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gold",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gold",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gold",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "paddingLeft": 10,
            "paddingRight": 10,
            "textAlign": "center",
          }
        }>
         
        ¿Qué nota le pondría a la tienda en general?
         
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gold",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gold",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
        <View
          accessibilityComponentType="button"
          accessibilityLabel={undefined}
          accessibilityTraits="button"
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "gray",
                  "fontFamily": "Ionicons",
                  "fontSize": 40,
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                undefined,
              ]
            }>
            
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginLeft": 15,
          "marginRight": 15,
          "marginTop": 15,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "paddingLeft": 10,
            "paddingRight": 10,
            "textAlign": "center",
          }
        }>
         Tu comentario 
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "marginTop": 10,
            },
            Object {
              "fontSize": 16,
              "textAlign": "center",
            },
          ]
        }>
         
      </Text>
    </View>
  </ScrollView>
</View>
`;
